lib/src/usage_impl.dart

```diff
@@ -75,17 +81,32 @@ class AnalyticsImpl implements Analytics {
   @override
   AnalyticsOpt analyticsOpt = AnalyticsOpt.optOut;
 
+  late Future<void>? Function() _batchingDelay;
+  final Queue<String> _batchedEvents = Queue<String>();
+  bool _isSendingScheduled = false;
+
   late final String _url;
+  late final String _batchingUrl;
 
   final StreamController<Map<String, dynamic>> _sendController =
       StreamController.broadcast(sync: true);
 
-  AnalyticsImpl(this.trackingId, this.properties, this.postHandler,
-      {this.applicationName, this.applicationVersion, String? analyticsUrl}) {
+  AnalyticsImpl(
+    this.trackingId,
+    this.properties,
+    this.postHandler, {
+    this.applicationName,
+    this.applicationVersion,
+    String? analyticsUrl,
+    String? analyticsBatchingUrl,
+    Future<void>? Function()? batchingDelay,
```

Comment:
Why use a callback function here instead of a `Duration` parameter?

Reply:
I thought it was more flexible - with a function returning a Future, the delay can relate to other events in the system (sending batched requests after a certain event has happened) but perhaps a Duration would be simpler and suffice for most use-cases...?

Reply:
@jonasfj convinced me this is indeed overkill - taking a `Duration` now!